guide/src/function.md

```diff
@@ -234,13 +211,12 @@ with only positional args.
 
 ### Calling Rust functions in Python
 
-If you have a static function, you can expose it with `#[pyfunction]` and use [`wrap_pyfunction!`]
-to get the corresponding [`PyCFunction`]. For dynamic functions, e.g. lambdas and functions that
-were passed as arguments, you must put them in some kind of owned container, e.g. a `Box`.
-(A long-term solution will be a special container similar to wasm-bindgen's `Closure`). You can
-then use a `#[pyclass]` struct with that container as a field as a way to pass the function over
-the FFI barrier. You can even make that class callable with `__call__` so it looks like a function
-in Python code.
+The ways to convert a Rust function into a Python object vary depending on the function:
+
+- Concrete functions, e.g. `fn foo()`: add `#[pyfunction]` and then use [`wrap_pyfunction!`] to get the corresponding [`PyCFunction`].
+- Dynamic functions, e.g. `foo: fn()` either:
```

Comment:
What about "named" and "anonymous/closures" versus "concrete" and "dynamic"? I don't think many people are familiar with those terms.

Reply:
Much better